src/text.h

```diff
@@ -410,12 +410,22 @@ struct Text {
     }
 
     void ReplaceStr(const wxString &str) {
-        for (int i = 0, j; (j = t.Mid(i).Lower().Find(sys->searchstring)) >= 0;) {
+        if(sys->casesensitivesearch) {
+            for (int i = 0, j; (j = t.Mid(i).Find(sys->searchstring)) >= 0;) {
             // does this need WasEdited()?
             i += j;
             t.Remove(i, sys->searchstring.Len());
             t.insert(i, str);
             i += str.Len();
+            }
+        } else {
+            for (int i = 0, j; (j = t.Mid(i).Lower().Find(sys->searchstring.Lower())) >= 0;) {

```

Comment:
the only difference between these two loops is whether the search string is lowered? why don't you do that out of the loop, or at the start of search? These are not cheap operations

Reply:
Thanks for the hints! I found another way to these operations before the loop. Thanks :-)